b2/b2http.py

```diff
@@ -117,6 +117,43 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         self.response.close()
 
 
+class HttpCallback(object):
+    """
+    A callback object that does nothing.  Override pre_request
+    and/or post_request as desired.
+    """
+
+    def pre_request(self, method, url, headers):
+        """
+        Called before processing an HTTP request.
+
+        Returns True if the request should be processed, and False
+        if the request should be aborted.
+
+        Raises an exception if this request should not be processed.
+        The exception raised must inherit from B2Error.
+
+        :param method: One of: 'POST', 'GET', etc.
+        :param url: The URL that will be used.
+        :param headers: The header sent with the request.
+
+        """
+
+    def post_request(self, method, url, headers, response):
+        """
+        Called after processing an HTTP request.
+        Should not raise an exception.
+
+        Raises an exception if this request should be treated as failing.
+        The exception raised must inherit from B2Error.
```

Comment:
It should inherit from `B2HttpCallbackPostRequestException`, which should inherit from `B2HttpCallbackException`, which should inherit from `B2Error` or `B2SimpleError`